src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -32,6 +32,11 @@ public static partial class Marshal
         internal static Guid IID_IUnknown = new Guid("00000000-0000-0000-C000-000000000046");
 #endif //FEATURE_COMINTEROP
 
+        static Marshal()
```

Comment:
An explicit static constructor is bad for performance and in this case also causes an infinite loop during startup. Maybe `nativeDllResolveMap` should be lazy-initialized on first use?

Reply:
Agree